### PR TITLE
Add Sharp decode and send functions.

### PR DESCRIPTION
**Pull request by SergiyKolesnikov**
The decode function correctly recognizes single button presses and
push-and-hold. The implementation does not require any changes to the core
library. It was tested with a Sharp GA825WJSA remote.

Related discussions: https://github.com/z3t0/Arduino-IRremote/pull/381 https://github.com/z3t0/Arduino-IRremote/issues/380

### CODE REVIEW

**Comment by SergiyKolesnikov** [comment]
Hi @z3t0, this PR has been pending for a long time. Are there any issues that prevent its merging? Can I some how help with them? I would really like to close it.

**Comment by z3t0** [comment]
Hey,

Sorry for the delay! I haven't been very active in this project. 

I'll take a look through and merge this ASAP.

**Comment by SergiyKolesnikov** [comment]
No worries and thanks for the time you have been investing in this.

File: ir_Sharp.cpp [hunk 1]
```diff
@@ -1,71 +1,162 @@
...
+//   http://www.sbprojects.com/knowledge/ir/sharp.htm
+//
+// Here is the LIRC file I found that seems to match the remote codes from the
+// oscilloscope:
+//   Sharp LCD TV:
+//   http://lirc.sourceforge.net/remotes/sharp/GA538WJSA
+
+#define SHARP_RAWLEN		32
+#define SHARP_ADDRESS_BITS	5
+#define SHARP_COMMAND_BITS	8
```
Comment (z3t0) [inline ir_Sharp.cpp]: @SergiyKolesnikov Just looking over the changes in the constants, it seems that the SHARP protocol you are writing is different to what was originally in the file?

The marks and spaces, seem to be different lengths.

I am concerned about existing projects that already use the previous "definition" of what the SHARP protocol is?

What are your thoughts on this?

Comment (z3t0) [reply ir_Sharp.cpp]: I would't mind merging your changes into a different namespace? I am just concerned about breaking existing projects

Comment (SergiyKolesnikov) [reply ir_Sharp.cpp]: Yes, using a different namespace is probably the best solution to preserve compatibility.

Comment (z3t0) [reply ir_Sharp.cpp]: Okay, cool.

If you would like to rename it to a different namespace, then that would be awesome. 

If you're busy that's alright as well, I'll try to get around to it eventually.

Comment (SergiyKolesnikov) [reply ir_Sharp.cpp]: Do you have any suggestion for the name? Would something like `sharp_alt` for "alternative" be OK?

Comment (z3t0) [reply ir_Sharp.cpp]: Yes, that would be perfect

Comment (SergiyKolesnikov) [reply ir_Sharp.cpp]: @z3t0 I moved my implementation to a different namespace as discussed and restored the old Sharp code. Please take a look.

Comment (SergiyKolesnikov) [reply ir_Sharp.cpp]: Hi @z3t0, hope you are well. Please take a look at this PR, so that we can close it. Thanks!

Comment (SergiyKolesnikov) [reply ir_Sharp.cpp]: Hi @z3t0, thought may be you have a minute and we can close the PR?

**Pull request closed without merging**